extensions/data-transfer/portability-data-transfer-flickr/src/main/java/org/datatransferproject/datatransfer/flickr/media/FlickrMediaImporter.java

```diff
@@ -0,0 +1,267 @@
+    private final RateLimiter perUserRateLimiter;
+
+    public FlickrMediaImporter(
+            AppCredentials appCredentials,
+            TemporaryPerJobDataStore jobStore,
+            Monitor monitor,
+            TransferServiceConfig serviceConfig) {
+        this.jobStore = jobStore;
+        this.flickr = new Flickr(appCredentials.getKey(), appCredentials.getSecret(), new REST());
+        this.uploader = flickr.getUploader();
+        this.imageStreamProvider = new ImageStreamProvider();
+        this.photosetsInterface = flickr.getPhotosetsInterface();
+        this.monitor = monitor;
+        this.perUserRateLimiter = serviceConfig.getPerUserRateLimiter();
+    }
+
+    @VisibleForTesting
+    FlickrMediaImporter(
+            Flickr flickr,
+            TemporaryPerJobDataStore jobstore,
+            ImageStreamProvider imageStreamProvider,
+            Monitor monitor,
+            TransferServiceConfig serviceConfig) {
+        this.flickr = flickr;
+        this.imageStreamProvider = imageStreamProvider;
+        this.jobStore = jobstore;
+        this.uploader = flickr.getUploader();
+        this.photosetsInterface = flickr.getPhotosetsInterface();
+        this.monitor = monitor;
+        this.perUserRateLimiter = serviceConfig.getPerUserRateLimiter();
+    }
+
+    @Override
+    public ImportResult importItem(
+            UUID jobId,
+            IdempotentImportExecutor idempotentExecutor,
+            AuthData authData,
+            MediaContainerResource data)
+            throws Exception, IOException {
+        Auth auth;
+        try {
+            auth = FlickrUtils.getAuth(authData, flickr);
+        } catch (FlickrException e) {
+            return new ImportResult(e);
+        }
+        RequestContext.getRequestContext().setAuth(auth);
+
+        Preconditions.checkArgument(
+                data.getAlbums() != null || data.getPhotos() != null, "Error: There is no data to import");
+
+        if (data.getAlbums() != null) {
+            storeAlbums(jobId, data.getAlbums());
+        }
+
+        if (data.getPhotos() != null) {
+            for (PhotoModel photo : data.getPhotos()) {
+                try {
+                    importSinglePhoto(idempotentExecutor, jobId, photo);
+                } catch (FlickrException e) {
+                    if (e.getMessage().contains("Upload limit reached")) {
+                        throw new DestinationMemoryFullException("Flickr destination memory reached", e);
+                    } else if (e.getMessage().contains("Photo already in set")) {
+                        // This can happen if we got a server error on our end, but the request went through.
+                        // When our retry strategy kicked in the request was complete and the photo already
+                        // uploaded
+                        continue;
+                    }
+                    throw new IOException(e);
+                }
+            }
+        }
+
```

Comment:
Can you add a comment TODO to implement the videos portion of this?

Reply:
I added a TODO

Reply:
did you push your latest commit? I don't see a TODO anywhere (or did you actually resolve the TODO already)?

Reply:
That TODO has been resolved, that was from before I added video to the importer